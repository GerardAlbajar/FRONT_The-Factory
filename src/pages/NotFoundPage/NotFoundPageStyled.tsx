import styled from "styled-components";

const NotFoundPageStyled = styled.section`
  background-image: linear-gradient(0deg, rgba(0, 0, 0, 1), rgba(0, 0, 0, 0.6)),
    url("images/background.jpg");
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center center;
  display: flex;
  flex-flow: column;
  height: 100%;
  overflow: auto;

  padding: 50px;
  text-align: left;

  color: #fff;

  header {
    margin: 0px;
    margin-bottom: 50px;
    img {
      width: 200px;
    }
  }

  p {
    font-size: 100px;
  }
`;

export default NotFoundPageStyled;
