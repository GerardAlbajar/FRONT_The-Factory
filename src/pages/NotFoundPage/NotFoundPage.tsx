import { NavLink } from "react-router-dom";
import NotFoundPageStyled from "./NotFoundPageStyled";

const NotFoundPage = (): JSX.Element => {
  return (
    <NotFoundPageStyled>
      <header>
        <NavLink to="/home">
          <img src="images/logo.png" alt="Astro Factory Logo" />
        </NavLink>
      </header>
      <p>Page Not Found</p>
    </NotFoundPageStyled>
  );
};

export default NotFoundPage;
