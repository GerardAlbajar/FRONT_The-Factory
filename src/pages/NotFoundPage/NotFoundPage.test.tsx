import { BrowserRouter } from "react-router-dom";
import NotFoundPage from "./NotFoundPage";
const { render, screen } = require("@testing-library/react");

describe("Given a HomePage Component", () => {
  describe("When it's invoked)", () => {
    test("Then it should render a the HomePage component with two buttons elements", () => {
      const expectedText = "Page Not Found";

      render(
        <BrowserRouter>
          <NotFoundPage />
        </BrowserRouter>
      );

      const selectedText = screen.getByText(expectedText);

      expect(selectedText).toBeInTheDocument();
    });
  });
});
