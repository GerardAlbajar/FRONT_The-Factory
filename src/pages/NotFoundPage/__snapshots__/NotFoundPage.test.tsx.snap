// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a HomePage Component When it's invoked) Then it should always match with this snapshot 1`] = `
<section
  className="sc-bczRLJ kvfDHZ"
>
  <header>
    <a
      className=""
      href="/home"
      onClick={[Function]}
    >
      <img
        alt="Astro Factory Logo"
        src="images/logo.png"
      />
    </a>
  </header>
  <p>
    Page Not Found
  </p>
</section>
`;
